internal/provider/resource_tfe_registry_module_test.go

```diff
@@ -803,6 +803,27 @@ func TestAccTFERegistryModule_invalidWithRegistryNameAndNoModuleProvider(t *test
 		},
 	})
 }
+
+func TestAccTFERegistryModule_deleteModuleProvider(t *testing.T) {
+	org, _ := setupDefaultOrganization(t)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckTFERegistryModuleDestroy,
```

Comment:
A question that has come up is whether we need to build a test case where  resources are destroyed if we have `CheckDestroy`?